optaplanner-examples/src/main/java/org/optaplanner/examples/tsp/domain/location/RoadLocationKeyDeserializer.java

```diff
@@ -0,0 +1,17 @@
+package org.optaplanner.examples.tsp.domain.location;
+
+import org.optaplanner.examples.tsp.persistence.TspSolutionFileIO;
+
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.KeyDeserializer;
+
+/**
+ * @see TspSolutionFileIO
+ */
+final class RoadLocationKeyDeserializer extends KeyDeserializer {
+
+    @Override
+    public Object deserializeKey(String value, DeserializationContext deserializationContext) {
+        return new RoadLocation(Long.parseLong(value)); // Need to be de-duplicated.
```

Comment:
Maybe one last thing we can try to avoid duplicate instances (haven't tried yet):
https://stackoverflow.com/questions/55924605/jackson-how-to-make-jsonbackreference-and-a-custom-deserializer-work-at-the-sa
Even if that works, we will need access to the `JacksonUniqueIdGenerator` in the `RoadLocationKeySerializer`.